discovery-provider/src/tasks/aggregates/index_aggregate_track.py

```diff
@@ -0,0 +1,138 @@
+import logging
+
+
+from src.tasks.aggregates import try_updating_aggregate_table, update_aggregate_table
+from src.tasks.celery_app import celery
+from src.tasks.index_aggregate_user import get_latest_blocknumber
+
+logger = logging.getLogger(__name__)
+
+AGGREGATE_TRACK = "aggregate_track"
+
+# UPDATE_AGGREGATE_TRACK_QUERY
+# Get a lower bound blocknumber to check for new save/repost activity
+# Find a subset of tracks that have changed after that blocknumber
+# For that subset of tracks calculate the play/repost total counts
+# Insert that count for new users or update an existing row
+UPDATE_AGGREGATE_TRACK_QUERY = """
+    WITH new_track AS (
+        SELECT
+            t.track_id AS track_id
+        FROM
+            tracks t
+        GROUP BY
```

Comment:
I have this unstaged commit from our call:

```
        WHERE
            t.is_current IS TRUE
            AND t.blocknumber > :prev_blocknumber
```

Reply:
I don't think we should ever check `t.is_current`, much less `t.blocknumbers`.

We want to join _all_ tracks with tracks that have been _reposted_ since `:prev_blocknumber`.

The way I'm reading the unstaged commit is that only songs that have updated their metadata after `:prev_blocknumber` will be eligible to have their counts updated if reposts have also occurred after `:prev_blocknumber`.

\* **reposts** above being both repost/save events

Reply:
So in this case, we actually do want this filtering. I think you may be confusing joining with union. We do want to consider any tracks that have changed after the blocknumber, for example if a track was deleted and is not longer is_current true. We also want to union that with any changes in reposts or saves since the block as well. After that, we recount each of those recently changed tracks.

Reply:
Couple questions for @dmanjunath: 

Discussed more with @joaquincasares, we think we might not need to select from tracks at all. Maybe union with reposts / saves should be good enough. We only care if those counts changed. A change in track should not matter.

Also, aggregate_track and aggregate_user mat views differ in one way where aggregate_track does not include deleted tracks and aggregate_user includes deleted users. In our current table updating, we follow aggregate_user's method so we never delete rows. I wonder how important that is.

Reply:
Ooh yeah that's true. We still do need select from tracks. That leaves just the second part where we probably would want another query that checks for deleted tracks and removes them from aggregate_track.

Reply:
Ah, good catch @dmanjunath . Thanks!

I've added the `AND t.blocknumber > :prev_blocknumber` back, but +1 towards Isaac's point that we now need to consider a delete tracks query or be okay with deleted tracks from having stale counts from before it was deleted. Would the latter cause any issues?

Reply:
lets ask @raymondjacobson or @jowlee about tracks that got deleted and if it should exist in the aggregate_tracks table. i'm not too worried about stuff being outdated because we shouldn't really have any social actions once a track gets deleted. but lets confirm with them if it should be in there or not.

Reply:
Yeah it is strange for aggregate_user and aggregate_track to differ in this way. It probably doesn't matter much.

Reply:
We shouldn't need to keep the tracks in the aggregate track table
Deleting it should be safe - I see it's only used in two places: `get_remixable_tracks` and `populate_track_metadata`
the remixables filter by deleted and populate track metadata uses it to get the number of reposts/saves and returns 0 for each of the tracks that do not exist - so it shouldn't error. 

From the UI side of things, it shouldn't affect anything. If a track is deleted we stop displaying it in most places: If you go to the deleted track page, we only show the title/artist name, and if a deleted track is in a playlist, we just show title, artist, time, date, and play count.

Reply:
cool thanks @jowlee. so sounds like we could do another CTE in addition to new_track where deleted_tracks AS (DELETE FROM "aggregate_tracks" WHERE "track_id" IN (SELECT "track_id" FROM "tracks" WHERE "blocknumber" > :prev_blocknumber AND "is_deleted" = TRUE)) @isaacsolo @joaquincasares

Reply:
since we either join to tracks or pass in track_ids i think it's okay to keep. i'll defer to @jowlee

Reply:
It's fine to keep it too - might be simpler that way. 
Echoing dheeraj's point, there shouldn't be any more social actions with that track id anyways.
Also worth noting that users now have a "is_deactivated" column - similar to the track's deleted field.